sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala

```diff
@@ -388,18 +388,13 @@ case class ArraySort(
     checkArgumentDataTypes() match {
       case TypeCheckResult.TypeCheckSuccess =>
         argument.dataType match {
-          case ArrayType(dt, _) if RowOrdering.isOrderable(dt) =>
```

Comment:
Hm, what about the case like: `array_sort((x, y) => x == y)` where `x` and `y` are map?

Reply:
This is not working, as a map is (currently) not orderable.
With and without this patch this will fail with the following error:
```
> Seq((Array[Map[String, Int]](Map("a" -> 1), Map()), "x")).toDF("a", "b").selectExpr("array_sort(a)")
org.apache.spark.sql.AnalysisException: cannot resolve '(namedlambdavariable() < namedlambdavariable())' due to data type mismatch: LessThan does not support ordering on type map<string,int>; line 1 pos 0;
```
Because this error message is emitted before the code flow can reach the logic which is removed in this patch.
So as far as I can see, the removed logic can currently only be hit in cases where it should work, hence the removal.

Reply:
Doesn't the above cast fail because there is `LessThan` used in the default comparator? @HyukjinKwon's case provides a lambda function. As you remove the check, the lambda will be executed.

Reply:
Your interpretation is correct.  `LessThan` fails for lambda's like this when x and y are structs.
`array_sort` does not accept `x > y` because it is not an integer function, but that is a detail here I think.

Reply:
Your interpretation is correct.  `LessThan` fails for lambda's like this when x and y are structs.
`array_sort` does not accept `x > y` because it is not an integer function, but that is a detail here I think.

Reply:
Added extra comment at defaultComparator to explain.